### PR TITLE
[MISC] Restore Codecov and update libs

### PR DESCRIPTION
**Pull request by dragomirp**
* Restore codecov and add configuration
* Update charm libs

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+github_checks:
+  annotations: false
```
Comment (dragomirp) [inline .github/codecov.yml:2]: Disable annotations

File: .github/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+coverage:
+  status:
+    project:
+      default:
+        target: 70%
```
Comment (dragomirp) [inline .github/codecov.yml:7]: Overall coverage should be over 70%

File: .github/codecov.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    patch:
+      default:
+        target: 33%
```
Comment (dragomirp) [inline .github/codecov.yml:10]: Commit delta should be at least 33% covered.

File: tests/unit/test_charm.py
Comment (dragomirp) [inline tests/unit/test_charm.py:1]: Log messages are slightly different in the latest lib.

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**